### PR TITLE
Update Play root span only if there is one

### PR DESCRIPTION
**Pull request by bantonsson**
Not sure that this will ever happen in a real world setting, but it logs exceptions while running the `play` test suite with agent debug logging enabled.

### CODE REVIEW

**Comment by richardstartin** [comment]
Could we put a single null check in to the `onRequest` method instead?

**Review by richardstartin — APPROVED** [review]

File: dd-java-agent/instrumentation/play-2.3/src/main/java8/datadog/trace/instrumentation/play23/PlayAdvice.java [hunk 1]
```diff
@@ -66,7 +66,9 @@ public static void stopTraceOnResponse(
     // span finished in RequestCompleteCallback
 
     final AgentSpan rootSpan = activeSpan();
-    // set the resource name on the upstream akka/netty span
-    DECORATE.onRequest(rootSpan, req);
+    // set the resource name on the upstream akka/netty span if there is one
+    if (rootSpan != null) {
+      DECORATE.onRequest(rootSpan, req);
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/play-2.3/src/main/java8/datadog/trace/instrumentation/play23/PlayAdvice.java:71]: This null check already happens inside the `onRequest` call.  I don't think this PR is necessary.

Comment (richardstartin) [reply dd-java-agent/instrumentation/play-2.3/src/main/java8/datadog/trace/instrumentation/play23/PlayAdvice.java:71]: That's inaccurate. The first call in `PlayHttpServerDecorator.onRequest` is to `HttpServerDecorator.onRequest` which asserts on the first line that the span is not null. On a more general note, the description of the PR states that this error was encountered running play's test suite, and getting a clean run on the Play 2.8 test suite was the motivation for doing the work.

Comment (tylerbenson) [reply dd-java-agent/instrumentation/play-2.3/src/main/java8/datadog/trace/instrumentation/play23/PlayAdvice.java:71]: oops. sorry, I misread the null check in `onRequest`.  Carry on!

**Review by tylerbenson — APPROVED** [review]

**Pull request merged**